### PR TITLE
[XPU]eliminate small pattern

### PR DESCRIPTION
**Pull request by wz1qqx**
### PR types

Performance optimization
### PR changes

OPs
### Description

- fold unsqueeze2+gather+squeeze2 to gather
- fold transpose+unsqueeze2+reduce_sum+act+transpose to act
- fix add_layernorm_fuse_pass as review

### CODE REVIEW

**Review by jiweibo — DISMISSED** [review]

**Review by jiweibo — APPROVED** [review]

File: paddle/phi/kernels/fusion/xpu/add_layernorm_xpu_kernel.cc [hunk 1]
```diff
@@ -37,12 +83,19 @@ void AddLayernormXPUKernel(const Context& ctx,
...
   float* mean_data = ctx.template Alloc<float>(mean);
   float* variance_data = ctx.template Alloc<float>(variance);
   auto* z_add_data = reinterpret_cast<XPUType*>(ctx.template Alloc<T>(z_add));
 
+  auto x_dims = x.dims();
+  auto y_dims = y.dims();
+  auto out_dims = BroadCastInferShape(x_dims, y_dims, -1);
+  auto layer_norm_x_mat_dims = phi::flatten_to_2d(out_dims, begin_norm_axis);
+  int64_t m = layer_norm_x_mat_dims[0];
+  int64_t n = layer_norm_x_mat_dims[1];
```
Comment (zhupengyang) [inline paddle/phi/kernels/fusion/xpu/add_layernorm_xpu_kernel.cc:95]: 这里不需要重新计算 out_dims，可以直接从 out 去拿

File: paddle/phi/kernels/fusion/xpu/add_layernorm_xpu_kernel.cc [hunk 2]
```diff
@@ -66,5 +119,4 @@ PD_REGISTER_KERNEL(add_layernorm_xpu,
                    XPU,
                    ALL_LAYOUT,
                    phi::fusion::AddLayernormXPUKernel,
-                   float,
-                   phi::dtype::float16) {}
+                   float) {}
```
Comment (zhupengyang) [inline paddle/phi/kernels/fusion/xpu/add_layernorm_xpu_kernel.cc:122]: 为什么不支持 fp16 了呢

**Review by zhupengyang — APPROVED** [review]
先合入，下个PR改下

File: paddle/phi/kernels/fusion/xpu/add_layernorm_xpu_kernel.cc (hunk 1 above)
Comment (wz1qqx) [reply paddle/phi/kernels/fusion/xpu/add_layernorm_xpu_kernel.cc:95]: 👌

**Pull request merged**